app/controllers/nodes_controller.rb

```diff
@@ -18,6 +18,42 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  # rubocop:disable Metrics/AbcSize
+  def assign
+    @minion = Minion.find(params[:node_id])
+
+    unless Minion.roles.keys.include?(params[:role])
+      raise Minion::InvalidRole, "Could not assign role #{params[:role]}"
+    end
+
+    respond_to do |format|
+      if @minion.assign_role(params[:role])
+        res, body = Velum::Salt.orchestrate
+        http_status = res.code == "200" ? :ok : :unprocessable_entity
+        flash[:error] = "Salt failed to orchestrate: #{body.inspect}" unless http_status == :ok
+
+        format.html { redirect_to nodes_path }
+        format.json { head http_status }
+      else
+        format.html do
+          flash[:error] = @minion.errors.full_messages.first
+          redirect_to nodes_path
+        end
+        format.json { render json: @minion.errors, status: :unprocessable_entity }
+      end
+    end
+  rescue Velum::SaltApi::SaltConnectionException,
+         ActiveRecord::RecordNotFound,
```

Comment:
i have tried to change that, but rubocop gave me a `Metrics/CyclomaticComplexity` violation. I think it's because of the additional `respond_to` block which i wanted to avoid, as i have stated below.

```ruby
  def assign
    if Minion.exists?(id: params[:node_id])
      @minion = Minion.find(params[:node_id])
    else
      error_msg = "Cannot assign role to non-existing Minion with id: #{params[:node_id]}"
      respond_to do |format|
        format.html do
          flash[:error] = error_msg
          redirect_to nodes_path && return
        end
        format.json { render json: error_msg, status: :unprocessable_entity && return }
      end
    end
    ...
  end
```

I can try to reduce the complexity further, by saving some errors in a hash or variables, and only have one intelligent `respond_to` block but is it really worth adding this complexity?